cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2383,6 +2394,35 @@ int getPodIndexFromPvcName(String pvcName)  {
             }
         }
 
+        Future<ReconciliationState> entityOperatorReady() {
+            long pollingIntervalMs = 1_000;
+            long timeoutMs = operationTimeoutMs;
+
+            if (this.entityOperator != null && eoDeployment != null) {
+                Future<Void> future = podOperations.readiness(namespace, this.entityOperator.getName(), 1_000, timeoutMs);
```

Comment:
The pod name will not be the same as the deployment name. I think you need to first get the pod name created by the deployment and only then run the `readiness` call.